proposals/open-telemetry/microprofile-opentelemetry-tracing.asciidoc

```diff
@@ -0,0 +1,190 @@
+:authors: Roberto Cortez
+:email: radcortez@yahoo.com
+:version-label!:
+:sectanchors:
+:doctype: book
+:license: Eclipse Foundation Specification License v1.0
+:source-highlighter: coderay
+:toc: left
+:toclevels: 4
+:sectnumlevels: 4
+ifdef::backend-pdf[]
+:pagenums:
+endif::[]
+
+= Eclipse MicroProfile OpenTelemetry Tracing
+
+== Introduction
+
+TODO
+
+== Rationale
+
+TODO
+
+== Architecture
+
+This specification defines an easy way to allow an application running in a MicroProfile container to take advantage of
+distributed tracing by using the https://opentelemetry.io[OpenTelemetry project] (a merger of the OpenTracing and
+OpenCensus projects).
+
+This document and implementations *MUST* comply with the following OpenTelemetry specifications:
+
+- https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/overview.md[OpenTelemetry Overview]
+(except for Metrics)
+- https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/api.md[Tracing API]
+- https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/baggage/api.md[Baggage API]
+- https://github.com/open-telemetry/opentelemetry-specification/tree/main/specification/context[Context API]
+- https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/resource/sdk.md[Resource SDK]
+
```

Comment:
Not sure if this adds any value because the modes are added next. Plus, we need to discuss if we will support a third mode: Agent Instrumentation.

Reply:
We discussed on today's call. For the agent instrumentation, this might be implementation detail. For the 1st release, we should keep it simple for now.

Reply:
While we can omit it from the spec, there is not much to be done at the spec level than to just say that MP OTel needs to behave nicely if an OTel Agent is plugged in and probably require to not double the spans added by the Agent.

OTel Agent instrumentation is the most accepted way to add OTel integration into your applications. Typical monitoring tools like New Relic or Datadog rely on the Agent to transparently monitor applications without requiring code changes. I believe this is a too big use case to ignore.

Reply:
Will handle this in a separate PR.